rust-runtime/aws-smithy-json/src/deserialize/token.rs

```diff
@@ -168,7 +164,7 @@ pub fn expect_number_or_null(
         Some(Token::ValueNumber { value, .. }) => Ok(Some(value)),
         Some(Token::ValueString { value, offset }) => match value.to_unescaped() {
             Err(err) => Err(Error::new(
-                ErrorReason::Custom(format!("expected a valid string, escape was invalid: {}", err).into()), Some(offset.0))
+                ErrorKind::Custom(format!("expected a valid string, escape was invalid: {}", err).into()), Some(offset.0))
```

Comment:
Same here - should this `err` be captured as source?

Reply:
I think I initially avoided this to avoid fixing all the tests. You'll see what I mean with the new commit I pushed that keeps the source around.

Reply:
Yeah, I see what you mean now.